sql/json.go

```diff
@@ -78,10 +78,12 @@ func (t jsonType) SQL(dest []byte, v interface{}) (sqltypes.Value, error) {
 		return sqltypes.NULL, nil
 	}
 
-	js, ok := v.(JSONValue)
-	if !ok {
+	// Convert to jsonType
+	jsVal, err := t.Convert(v)
```

Comment:
Change looks good. Anything we can add to `json_test.go` that would have caught this for us?